tests/integration/test_advanced_indexing.py

```diff
@@ -81,6 +95,98 @@ def test_setitem_scalar_1d(arr, idx, val):
     assert np.array_equal(arr, [-1])
 
 
+@pytest.mark.parametrize("idx", IDXS_EMPTY_1D)  # idx = []
+@pytest.mark.parametrize("arr", ARRS_EMPTY_1D)  # arr = [42], [5], []
+def test_getitem_empty_1d(arr, idx):
+    assert np.array_equal(arr[idx], [])
+
+
+@pytest.mark.parametrize("idx", IDXS_EMPTY_1D)  # idx = []
+@pytest.mark.parametrize("arr", ARRS_EMPTY_1D)  # arr = []
+@pytest.mark.parametrize("val", VALS_EMPTY_1D)  # val = []
+def test_setitem_empty_1d(arr, idx, val):
+    arr[idx] = val
+    assert np.array_equal(arr[idx], [])
+
+
+def mk_deferred_array(lib, shape):
+    if np.prod(shape) != 0:
+        return lib.ones(shape)
+    # for shape (2,0,3,4): good_shape = (2,1,3,4)
+    good_shape = tuple(max(1, dim) for dim in shape)
+    # for shape (2,0,3,4): key = [:,[False],:,:]
+    key = tuple([False] if dim == 0 else slice(None) for dim in shape)
+    print("IRINA DEBUG good_shape , key", good_shape, key)
+    return lib.ones(good_shape)[key]
+
+
+def test_zero_size():
```

Comment:
@bryevdv Is there a way to write this with fixtures, where the fixture for `arr_ndim` ranges independently from 1 to `LEGATE_MAX_DIM + 1`, and the fixtures for `idx_ndim` and `zero_dim` depend on the chosen value of `arr_ndim`?

Reply:
`pytest.mark.parametrize` can accept a callable that returns an iterable. So I'd suggest writing a little helper function that yields the desired argument tuples.

Reply:
thank you @bryevdv ! I have updated the test accordingly